src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,65 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendarGap;
+  max-width: $calendarSize;
+  padding: $calendarPadding;
+  margin-left: auto;
+  margin-right: auto;
+
+  $days:
+    "mon" 1,
+    "tue" 2,
+    "wed" 3,
+    "thu" 4,
+    "fri" 5,
+    "sat" 6,
+    "sun" 7;
+
+  @each $name, $dayNr in $days {
+    &--start-day-#{$name} &__day:first-child {
+      margin-left: ($daySize + $calendarGap) * ($dayNr - 1);
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} &__day:nth-last-of-type(-n + #{31 - $i}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    width: $daySize;
+    height: $daySize;
+    border: 1px solid black;
+    background-color: $calendarBg;
+    position: relative;
+    transition: all 0.5s ease-in;
+
+    &::before {
+      content: " ";
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+      display: flex;
+      position: absolute;
+      left: 0;
+      top: 0;
+      width: 100%;
+      height: 100%;
```

Comment:
Instead of positioning it absolutely, try to center it with flex added to __day.
Your solution works but it can be simpler ;)